In [ ]:
import pandas as pd
df = pd.read_csv("/content/drive/MyDrive/ML_practise_project/titanic.csv")
df.head()

PassengerId                                               Name  Pclass  \
0            1                            Braund, Mr. Owen Harris       3   
1            2  Cumings, Mrs. John Bradley (Florence Briggs Th...       1   
2            3                             Heikkinen, Miss. Laina       3   
3            4       Futrelle, Mrs. Jacques Heath (Lily May Peel)       1   
4            5                           Allen, Mr. William Henry       3   

      Sex   Age  SibSp  Parch            Ticket     Fare Cabin Embarked  \
0    male  22.0      1      0         A/5 21171   7.2500   NaN        S   
1  female  38.0      1      0          PC 17599  71.2833   C85        C   
2  female  26.0      0      0  STON/O2. 3101282   7.9250   NaN        S   
3  female  35.0      1      0            113803  53.1000  C123        S   
4    male  35.0      0      0            373450   8.0500   NaN        S   

   Survived  
0         0  
1         1  
2         1  
3         1  
4         0

In [ ]:
df.drop(["PassengerId", "Name", "SibSp", "Parch", "Ticket", "Cabin", "Embarked"], axis="columns", inplace=True)
df.head()

Pclass     Sex   Age     Fare  Survived
0       3    male  22.0   7.2500         0
1       1  female  38.0  71.2833         1
2       3  female  26.0   7.9250         1
3       1  female  35.0  53.1000         1
4       3    male  35.0   8.0500         0

In [ ]:
target=df.Survived
inputs=df.drop("Survived", axis="columns")

In [ ]:
dummies = pd.get_dummies(inputs.Sex)
dummies.head(5)

female  male
0       0     1
1       1     0
2       1     0
3       1     0
4       0     1

In [ ]:
inputs = pd.concat([inputs,dummies],axis="columns")
inputs.head()

Pclass     Sex   Age     Fare  female  male
0       3    male  22.0   7.2500       0     1
1       1  female  38.0  71.2833       1     0
2       3  female  26.0   7.9250       1     0
3       1  female  35.0  53.1000       1     0
4       3    male  35.0   8.0500       0     1

In [ ]:
inputs.drop("Sex", axis="columns",inplace=True)
inputs.head()    

Pclass   Age     Fare  female  male
0       3  22.0   7.2500       0     1
1       1  38.0  71.2833       1     0
2       3  26.0   7.9250       1     0
3       1  35.0  53.1000       1     0
4       3  35.0   8.0500       0     1

In [ ]:
inputs.columns[inputs.isna().any()]

Index(['Age'], dtype='object')

In [ ]:
inputs.Age[:10]

0    22.0
1    38.0
2    26.0
3    35.0
4    35.0
5     NaN
6    54.0
7     2.0
8    27.0
9    14.0
Name: Age, dtype: float64

In [ ]:
inputs.Age = inputs.Age.fillna(inputs.Age.mean())
inputs.head(8)

Pclass        Age     Fare  female  male
0       3  22.000000   7.2500       0     1
1       1  38.000000  71.2833       1     0
2       3  26.000000   7.9250       1     0
3       1  35.000000  53.1000       1     0
4       3  35.000000   8.0500       0     1
5       3  29.699118   8.4583       0     1
6       1  54.000000  51.8625       0     1
7       3   2.000000  21.0750       0     1

In [ ]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(inputs,target,test_size=0.2)

In [ ]:
len(inputs)

891

In [ ]:
len(X_train)

712

In [ ]:
len(X_test)

179

In [ ]:
from sklearn.naive_bayes import GaussianNB
model = GaussianNB()

In [ ]:
model.fit(X_train,y_train)

GaussianNB()

In [ ]:
model.score(X_test,y_test)

0.8044692737430168

In [ ]:
y_test[:10]

606    0
345    1
860    0
30     0
813    0
367    1
702    0
838    1
495    0
691    1
Name: Survived, dtype: int64

In [ ]:
model.predict(X_test[:10])

array([0, 1, 0, 0, 1, 1, 1, 0, 0, 1])

In [ ]:
model.predict_proba(X_test[:10])

array([[0.99065548, 0.00934452],
       [0.03382612, 0.96617388],
       [0.99076786, 0.00923214],
       [0.91774176, 0.08225824],
       [0.04335   , 0.95665   ],
       [0.07917801, 0.92082199],
       [0.06874725, 0.93125275],
       [0.98176205, 0.01823795],
       [0.99083855, 0.00916145],
       [0.0443322 , 0.9556678 ]])

According the result 0.99065548% didn't survive so the value is 0 and 0.00934452% survived so the value is 1